pandas/tests/sparse/test_series.py

```diff
@@ -88,6 +88,18 @@ def setup_method(self, method):
         self.ziseries2 = SparseSeries(arr, index=index, kind='integer',
                                       fill_value=0)
 
+    def test_constructor_data_input(self):
+        # see gh-16905
+        test_dict = {1: 1}
+        test_index = [0, 1, 2]
+        test_series = pd.Series(test_dict, index=test_index)
+
+        arr = SparseSeries(test_dict)
+        assert arr.count() == len(test_dict)
+
+        arr = SparseSeries(test_series, index=test_index)
+        assert arr.count() == test_series.count()
```

Comment:
The second test is testing the creation of a SparseSeries from a Series. The SparseSeries constructor takes either a dict or a Series. Aside from class, the SparseSeries should be equal to the original Series.